### PR TITLE
Replace match_arglist in .set() with handwritten code instead

### LINKED ISSUE #52: WPI_TalonSRX.set adds 2ms overhead?
As reported by @ArthurAllshire .... need more details, and need to dig into this.

### PR DESCRIPTION
**Pull request by virtuald**
- Fixes #52
- Timeit shows on a roborio:
    match_arglist: 0.000162
    this impl:     0.000018

### CODE REVIEW

**Comment by auscompgeek** [comment]
Since the set signature changed in 5.3.1.0, why don't we just only special-case the 1-arg form here?

**Comment by virtuald** [comment]
Backwards compat?

File: ctre/wpi_talonsrx.py [hunk 1]
```diff
@@ -88,32 +87,32 @@ def set(self, *args, **kwargs):
...
+                demand0 = kwargs.pop('demand0', None)
+                demand1 = kwargs.pop('demand1', None)
+            else:
                 mode = ControlMode.PercentOutput
-            elif index == 1:
-                value = results['value']
-                mode = results['mode']
-            super().set(mode, value)
+                self.speed = demand0 = args[0]
+        else:
```
Comment (ariovistus) [inline ctre/wpi_talonsrx.py]: shouldn't set(mode=mode, demand0=1, demand1=0) be handled?

**Comment by auscompgeek** [comment]
This is less backwards-compatible with what it currently is anyway - this doesn't allow `set(mode=..., value=...)` and `set(speed=...)`. I suspect just special-casing the 1-arg form and having it passthrough otherwise would probably be more backwards-compat, ironically.

**Pull request merged**